src/toolchain.rs

```diff
@@ -431,6 +431,19 @@ impl Toolchain {
     }
 
     pub fn uninstall_self(&self) -> Result<()> {
+        let store = Store::from_env()?;
+        Components::collect_publishables()?
+            .into_iter()
+            .filter(|component| self.has_component(&component.name))
+            .filter_map(|component| {
+                get_bin_version(&self.bin_path.join(&component.name))
+                    .ok()
+                    .map(|version| store.component_dir_path(&component.name, &version))
+            })
+            .for_each(|p| {
+                let _ = remove_dir_all(p);
+            });
```

Comment:
This would be cleaner as a function of `Store`'s implementation so we can do `store.remove_component(&component.name, &version)`